client/src/components/Comments/Comments.jsx

```diff
@@ -1,112 +1,100 @@
-import React, { Component, Fragment } from "react";
-import PropTypes from "prop-types";
-import { NavLink as Link } from "react-router-dom";
-import _ from "lodash";
-import Comment from "./Comment";
-import AddComment from "./AddComment";
-import api from "../../services/api";
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux'
+import _ from 'lodash';
+import Comment from './Comment';
+import AddComment from './AddComment';
+import api from 'services/api';
+import 'assets/bootstrap.css';
+import { openModal } from 'redux/modal/slice'
+import './Comments.css';
 
-import "../../assets/bootstrap.css";
-import "./Comments.css";
+const Comments = props => {
+	const dispatch = useDispatch();
+	const [ state, setState ] = React.useState({
+		comments: [],
+		isLoading: true,
+		me: {},
+		error: false
+	})
 
-const signInURL = {
-    pathname: "/sign/in",
-    state: {
-        redirectTo: window.location.pathname
-    }
-};
-
-class Comments extends Component {
-    state = {
-        comments: [],
-        isLoading: true,
-        me: this.props.me || {},
-        error: false
-    };
-
-    componentDidMount() {
-        const { planId } = this.props;
-
-        return api
-            .get("/comment/" + planId)
-            .then(comments => {
-                this.setState({
-                    comments: comments.data,
-                    isLoading: false,
-                    me: comments.me
-                });
-            })
-            .catch(error => this.setState({ error }));
-    }
-
-    handleCommentPublished(data) {
-        var newComment = _.merge(data, {
-            person: {
-                alias: this.state.me.alias,
-                id: this.state.me.id
-            }
-        });
+	React.useEffect(() => {
+		const { planId } = props;
+		api.get('/comment/' + planId)
+			.then(comments => {
+				setState(pv => ({
+					...pv,
+					comments: comments.data,
+					isLoading: false,
+					me: comments.me
+				}));
+			})
+			.catch(error => setState(pv => ({ ...pv, error })));
+	}, [props])
 
-        // if there hasnt been an alias for current user
-        const newMe = this.state.me;
-        newMe.alias = newMe.alias || data.person.alias;
-        this.setState({
-            comments: this.state.comments.concat([newComment]),
-            content: "",
-            me: newMe
-        });
+	const handleCommentPublished = (data) => {
+		var newComment = _.merge(data, {
+			person: {
+				alias: state.me.name,
+				id: state.me.id
+			}
+		});
+		const newMe = state.me;
+		newMe.alias = newMe.alias || data.person.alias;
```

Comment:
Does the alias thing work properly? It should promote the alias input on users first comment, then save it